### ISSUE TITLE
Better naming of sub-packages

### ISSUE THREAD

**Issue by kormat** [scionproto/scion#1768]
Examples of very generic names:
- `go/lib/pathdb/conn`
- `go/lib/pathdb/query`
- `go/lib/pathdb/sqlite` (even clashes with `go/lib/sqlite`)
- `go/lib/overlay/conn`
- `go/lib/ctrl/extn`
- `go/lib/ctrl/ifid`
- `go/lib/ctrl/seg`
- `go/lib/infra/transport`

As an example, if a reader sees `conn.Conn` in a go file, that gives them zero idea what that might actually be. They have to consult the import list, on a per-file basis.

To put it another way, any package that's imported by something other than its parent should have an identifiable name. E.g. `go/lib/pathdb/conn` could be `go/lib/pathdb/pdbconn`. That way `pdbconn.Conn` is unambigious.

**Comment by kormat** [comment]
And, while we're at it, let's rename `go/lib/infra/disp` to perhaps `go/lib/infra/infradisp`, to reduce confusion with the scion dispatcher.

**Comment by scrye** [comment]
I agree that package names are not ideal right now. The golden rule for package naming should be https://github.com/golang/go/wiki/CodeReviewComments#package-names.

Since this is an ongoing effort (that is also mentioned in the style guide we reference in https://github.com/scionproto/scion/wiki/Coding-Guidelines), we should create individual issues for each package name that should be refactored. That will make it much easier to track than a catch-all issue. It will also make it easier to discuss what represents a good package name, because every instance of the naming problem is slightly different.

Regarding the suggestions in this issue, `pathdb/conn.Conn` should maybe become `pathdb.Conn` (similarly, `pathdb.Query`). The other packages will probably be going away altogether with the move to gRPC/standard UDP underlay sockets.

**Comment by ekuiter** [comment]
The coding guidelines document you linked to seems to be down?

**Comment by scrye** [comment]
> The coding guidelines document you linked to seems to be down?

Yes, sorry about that. We recently moved our documentation to a Read the Docs site. The new location for the Go style guide is [here](https://anapaya-scion.readthedocs-hosted.com/en/latest/contribute.html#go-style-guide).

**Comment by scrye** [comment]
The package name guidelines are [here](https://scion.docs.anapaya.net/en/latest/contribute.html#go-style-guide). Closing this because it's not actionable.
